### ISSUE TITLE
expand\collapse of group issue

### ISSUE THREAD

**Issue by sundertinwar** [mleibman/SlickGrid#1075]
I was working with grouping in slickgrid. When I expand and collapse groups I see that UI is not rendering properly and some rows are not visible at all.

http://jsfiddle.net/sunder_tiwnar/ywzkgwva/

http://mleibman.github.io/SlickGrid/examples/example-grouping
For example open any of link, go to last group expand it. Then expand group above it, then expand group above it. Carefully notice that some rows will not be visible. In first link you can notice that group ninth will go away. I can help to reproduce this bug in all browsers.

Any solution?

Thanks

**Comment by 6pac** [comment]
Could you test and see if there is still a problem with the version in my  ['alternative master' repo](https://github.com/6pac/SlickGrid) (also see [the wiki](https://github.com/6pac/SlickGrid/wiki)). I may have fixed this.
I notice that in your example, when I expand and collapse, I get two totals rows. I think that is what is 'eating' a regular row.

**Comment by sundertinwar** [comment]
Yes I can reproduce same with new repository. Here are snapshots in sequence number. I can follow below steps and reproduce it on any group level.
![1](https://cloud.githubusercontent.com/assets/666107/8653924/ce70d284-29a4-11e5-9847-8617edcb7c01.PNG)
![2](https://cloud.githubusercontent.com/assets/666107/8653928/d445c0ac-29a4-11e5-9927-e20047020161.PNG)
![3](https://cloud.githubusercontent.com/assets/666107/8653934/d9edbfe6-29a4-11e5-96cd-9e564d3250c7.PNG)
![4](https://cloud.githubusercontent.com/assets/666107/8653947/df68afa8-29a4-11e5-92e7-8be650aaaeff.PNG)
Let me know how it can be fixed?

**Comment by 6pac** [comment]
Thanks for trying that. I have a highly customised fork of SlickGrid (not the repo above) that I use for my own projects. I seem to recall finding and fixing a bug like this 18 months ago when I was rebuilding the grouping code. I'm in the process of trying to create a 'commit bridge' between the two forks now. I'll let you know if I can work it out, or if I find the fix in the other fork.

**Comment by sundertinwar** [comment]
Thanks, I appreciate your quick reply. Please do check with your private repository and let me know what can be fix for this issue. Keen to hear back from you.

**Comment by 6pac** [comment]
I notice that you're using the mleibman.github.io files for the example and in your jsfiddle.
I can reproduce the problem there, but that is not the main repo and could be an old code version.

Can you try downloading the main repo code (if you want jquery ~1.8) or my alternate master repo (if you want up to date jquery) and try locally.
I have copied your jsfiddle to a file in my local examples folder in my main repo and can't reproduce the problem. The elements are in the wrong order, but grouping works fine.

**Comment by 6pac** [comment]
This could well be due to the bug in the main trunk:

    function calculateTotals(totals) {
      var group = totals.group;
      var gi = groupingInfos[group.level];
      var isLeafLevel = (group.level == groupingInfos.length);
      var agg, idx = gi.aggregators.length;

      if (!isLeafLevel && gi.aggregateChildGroups) {
        // make sure all the subgroups are calculated
        var i = group.groups.length;
        while (i--) {
          if (!group.groups[i].initialized) {
            calculateTotals(group.groups[i]);
          }
        }
      }

the last section should read:

          if (!group.groups[i].totals.initialized) {
            calculateTotals(group.groups[i].totals);
          }

It's precisely bugs like this that made me put the 'alternative master' repo together.

**Comment by 6pac** [comment]
I have set up a Github pages site for my repo as well. Now you can check out the grouping example live: 
http://6pac.github.io/SlickGrid/examples/example-grouping
